File: 프로그래머스/unrated/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def getIndex(bias):
    if (bias == 'code'): return 0
    elif (bias == 'date'): return 1
    elif (bias == 'maximum'): return 2
    return 3

def solution(data, ext, val_ext, sort_by):
    answer = [[]]
    
    filtered = []
    idx = getIndex(ext)
    for d in data:
        if d[idx] < val_ext:
            filtered.append(d)
    idx = getIndex(sort_by)
    filtered.sort(key=lambda x: x[idx])
    
    return filtered

File: 프로그래머스/unrated/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;


class Solution {
    private int getIndex(String ext) {
        if (ext.equals("code")) {
            return 0;       
        } else if (ext.equals("date")) {
            return 1;
        } else if (ext.equals("maximum")) {
            return 2;
        } else {
            return 3;
        }
    }
    public ArrayList<int[]> solution(int[][] data, String ext, int val_ext, String sort_by) {
        int[][] answer = {};
        
        ArrayList<int[]> arr = new ArrayList<int[]>();
        for (int[] d : data) {
            int idx = this.getIndex(ext);
            if (d[idx] < val_ext) {
                arr.add(d);
            }
        }
        
        int sortIdx = this.getIndex(sort_by);
        arr.sort(new Comparator<int[]>() {
            @Override
            public int compare(int[] i1, int[] i2) {
                return i1[sortIdx] - i2[sortIdx];
            }
        });
        
        return arr;
    }
}